packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -170,11 +170,8 @@ class ImageCache {
       // Images that fail to load don't contribute to cache size.
       final int imageSize = info?.image == null ? 0 : info.image.height * info.image.width * 4;
       final _CachedImage image = _CachedImage(result, imageSize);
-      // If the image is bigger than the maximum cache size, and the cache size
-      // is not zero, then increase the cache size to the size of the image plus
-      // some change.
       if (maximumSizeBytes > 0 && imageSize > maximumSizeBytes) {
-        _maximumSizeBytes = imageSize + 1000;
+        throw 'Attempting to cache an image whose size exceeds ImageCache.maximumSizeBytes: $maximumSizeBytes';
```

Comment:
Prefer using Exception subtypes if it can/should be caught by users or Error subtypes if it is fatal

Reply:
It is almost impossible to catch, it's gobbled up here: https://github.com/flutter/flutter/blob/e60c0f68528d95d80783be58ba8b58cf49bce832/packages/flutter/lib/src/painting/image_stream.dart#L409:L409

I saw other people throwing strings in the framework.  Let me know what makes more sense.

Reply:
"Prefer" = please do this :)

Reply:
Ok, but it is technically neither of those things, it isn't catchable and it isn't fatal haha.

Reply:
(I didn't change this right away because we are still trying to figure out the best way to report this error, I'm not ignoring the comment =) . I'll pick this up after thanksgiving.